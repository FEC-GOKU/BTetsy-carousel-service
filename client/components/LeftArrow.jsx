import React from 'react';
import styles from './styles.css';

const LeftArrow = ({ prevPicture }) => (
  <div role="button" tabIndex="0" aria-label="left arrow" className={`${styles.leftArrow} ${styles.arrow}`} onClick={prevPicture}>
    <i className="fa fa-angle-left fa-3x" aria-hidden="true" />
  </div>
);

export default LeftArrow;
