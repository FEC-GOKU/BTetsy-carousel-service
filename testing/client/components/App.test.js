/**
 * @jest-environment jsdom
 */
import React from 'react';
import Enzyme from 'enzyme';
import { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({adapter: new Adapter()});
import App from '../../../client/components/app.jsx';
import CarouselPic from '../../../client/components/carousel.jsx';
import Display from '../../../client/components/display.jsx';
import LeftArrow from '../../../client/components/leftArrow.jsx';
import RightArrow from '../../../client/components/rightArrow.jsx';

describe('App', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<App debug />);
    expect(component).toMatchSnapshot();
  });
});

describe('<CarouselPic />', () => {
  test('It renders a picture based on prop', () => {

    let wrapper = Enzyme.mount(<CarouselPic image={'https://btetsy.s3.us-east-2.amazonaws.com/2_1.jpg'} />);

    expect(wrapper.find('.carouselPic').get(0).props.style.backgroundImage).toBe('url(https://btetsy.s3.us-east-2.amazonaws.com/2_1.jpg)');
  });
});

