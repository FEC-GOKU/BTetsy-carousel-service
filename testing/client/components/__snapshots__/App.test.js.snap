// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LeftArrow/> It renders a left button in the navaigation bar 1`] = `ReactWrapper {}`;

exports[`<RightArrow/> It renders a right button in the navaigation bar 1`] = `ReactWrapper {}`;

exports[`App should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;
